.changelog/11477.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+connect: add Namespace configuration entry for Vault provider
```

Comment:
smallest of nits: "configuration entry" is an overloaded term here, since Consul supports API objects named as such. Maybe "configuration setting" or just "configuration"